plugin/backend_lookup.go

```diff
@@ -110,19 +108,20 @@ func AAAA(b ServiceBackend, zone string, state request.Request, previousRecords
 			if dnsutil.DuplicateCNAME(newRecord, previousRecords) {
 				continue
 			}
-
-			state1 := state.NewWithQuestion(serv.Host, state.QType())
-			nextRecords, err := AAAA(b, zone, state1, append(previousRecords, newRecord), opt)
-
-			if err == nil {
-				// Not only have we found something we should add the CNAME and the IP addresses.
-				if len(nextRecords) > 0 {
-					records = append(records, newRecord)
-					records = append(records, nextRecords...)
+			if dns.IsSubDomain(zone, dns.Fqdn(serv.Host)) {
```

Comment:
Hmm, there were there before *shrug*

Reply:
@miekg The etcd unit test will failed when I remove dns.Fqdn. Because of "CNAME" target isn't end with '.'.
https://github.com/coredns/coredns/blob/75f1b9c988e036a56f0ae31224401aa9b652e9ef/plugin/etcd/cname_test.go#L53-L60